fusequery/query/src/datasources/datasource.rs

```diff
@@ -20,32 +20,14 @@ use crate::datasources::remote::RemoteDatabase;
 use crate::datasources::remote::RemoteFactory;
 use crate::datasources::remote::RemoteTable;
 use crate::datasources::system::SystemFactory;
-use crate::datasources::IDatabase;
-use crate::datasources::ITable;
-use crate::datasources::ITableFunction;
-
-#[async_trait::async_trait]
-pub trait IDataSource: Sync + Send {
```

Comment:
Is this necessary to support different source files, e.g., csv, table, parquet?

Reply:
Currently it is not going to support

Reply:
The name `DataSource` is somehow misleading, IMHO; The functionalities it provides are more like a catalog (e.g. CK's DatabaseCatalog) than a DataSource(especially in the context of Spark).

The reason for removing IDataSource and use a struct directly  :

> But datafuse have some conflicts after rename:
> trait IDataSource
> struct Datasource
> 
> If we rename IDataSource to DataSource, how to deal the struct name?
> 

_Originally posted by @BohuTANG in https://github.com/datafuselabs/datafuse/issues/727#issuecomment-854820280_


> Remove `IDataSource`, use `struct DataSource` directly? (replace `Arc<dyn IDatasource>` with `Arc<DataSource>`). 
> 
> Till we have multiple impls for an "IDataSource", and at that time we will have different names for impls naturally.

_Originally posted by @dantengsky in https://github.com/datafuselabs/datafuse/issues/727#issuecomment-854845547_

But, I am considering re-introduce the DataSouce/Catalog trait (but mainly for ease of testing), hopefully, there will be an PR in few days.